lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -225,15 +226,20 @@ public ContextRuntime SpawnChildContext(IConfiguration childContextConfiguration
                 {
                     Utilities.Diagnostics.Exceptions.Caught(e, Level.Error, LOGGER);
 
-                    Optional<string> parentId = ParentContext.IsPresent() ?
-                        Optional<string>.Of(ParentContext.Value.Id) :
-                        Optional<string>.Empty();
-                    ContextClientCodeException ex = new ContextClientCodeException(ContextClientCodeException.GetId(childContextConfiguration), parentId, "Unable to spawn context", e);
-                    
-                    Utilities.Diagnostics.Exceptions.Throw(ex, LOGGER);
+                    var contextId = string.Empty;
+                    try
+                    {
+                        var injector = TangFactory.GetTang().NewInjector(childContextConfiguration);
+                        contextId = injector.GetNamedInstance<ContextConfigurationOptions.ContextIdentifier, string>();
+                    }
+                    catch (InjectionException)
+                    {
+                        LOGGER.Log(Level.Error, "Unable to get Context ID from child ContextConfiguration. Using empty string.");
```

Comment:
minor: wondering why not to use Utilities.Diagnostics.Exceptions.Caught for consistancy (same as a above)